jupyterlab/labapp.py

```diff
@@ -375,35 +404,23 @@ class LabWorkspaceApp(JupyterApp):
     )
 
     def start(self):
-        super().start()
-        print('Either `export` or `import` must be specified.')
-        self.exit(1)
-
+        try:
+            super().start()
+            print('Either `export` or `import` must be specified.')
+            self.exit(1)
+        except NoStart:
+            pass
+        self.exit(0)
 
-lab_aliases = dict(aliases)
-lab_aliases['app-dir'] = 'LabApp.app_dir'
 
-lab_flags = dict(flags)
-lab_flags['core-mode'] = (
-    {'LabApp': {'core_mode': True}},
-    "Start the app in core mode."
-)
-lab_flags['dev-mode'] = (
-    {'LabApp': {'dev_mode': True}},
-    "Start the app in dev mode for running from source."
-)
-lab_flags['watch'] = (
-    {'LabApp': {'watch': True}},
-    "Start the app in watch mode."
-)
-lab_flags['expose-app-in-browser'] = (
-    {'LabApp': {'expose_app_in_browser': True}},
-    "Expose the global app instance to browser via window.jupyterlab"
-)
+class LabApp(NBClassicConfigShimMixin, LabServerApp):
+    version = version
 
+    extension_name = "jupyterlab"
```

Comment:
We should also change the `name` to `lab`, so our config is in `jupyter_lab_config.py` instead of `jupyter_notebook_config.py` by default.

Reply:
This is a tricky one... Changing the extension_name to lab fails on start (see stacktrace hereafter). There is a long history behind extension loading mechanism and I will call @Zsailer to the rescue here. I have tried a few things to make it work (changing _jupyter_server_extension_paths...) but no positive outcome so far... I have the impression that the extension_name is expected to match the python module name which should not...

And yes, the config filename `jupyter_jupyterlab_config.py` may be not the best

```
jupyter lab --generate-config
Writing default config to: /Users/echar4/.jupyter/jupyter_jupyterlab_config.py
```

```
Traceback (most recent call last):
  File "/opt/miniconda3/envs/datalayer/bin/jupyter-lab", line 11, in <module>
    load_entry_point('jupyterlab', 'console_scripts', 'jupyter-lab')()
  File "/Users/echar4/datalayer/repos/jupyter-server/jupyter_server/extension/application.py", line 426, in launch_instance
    load_other_extensions=cls.load_other_extensions
  File "/Users/echar4/datalayer/repos/jupyter-server/jupyter_server/extension/application.py", line 319, in initialize_server
    serverapp.initialize(argv=argv, load_extensions=load_other_extensions)
  File "<decorator-gen-7>", line 2, in initialize
  File "/opt/datalayer/opt/miniconda3/envs/datalayer/lib/python3.7/site-packages/traitlets/config/application.py", line 87, in catch_config_error
    return method(app, *args, **kwargs)
  File "/Users/echar4/datalayer/repos/jupyter-server/jupyter_server/serverapp.py", line 1783, in initialize
    self.init_server_extensions()
  File "/Users/echar4/datalayer/repos/jupyter-server/jupyter_server/serverapp.py", line 1541, in init_server_extensions
    mod, metadata_list = _get_server_extension_metadata(module_name)
  File "/Users/echar4/datalayer/repos/jupyter-server/jupyter_server/extension/serverextension.py", line 43, in _get_server_extension_metadata
    m = import_item(module)
  File "/opt/datalayer/opt/miniconda3/envs/datalayer/lib/python3.7/site-packages/traitlets/utils/importstring.py", line 42, in import_item
    return __import__(parts[0])
ModuleNotFoundError: No module named 'lab'
```

Reply:
Did a few more tests with the sever simple examples and opened https://github.com/jupyter/jupyter_server/issues/222 for discussion and solution definition.

Reply:
Ah, so with `jupyter_server`, `extension_name` is used in place of `name` to determine the config file path?

Reply:
Thanks for bringing this up, y'all! I went back and forth on this, so it's good to get some discussion around `extension_name`. 

> Ah, so with jupyter_server, extension_name is used in place of name to determine the config file path?

Yeah, that's right.

I was trying to consolidate the extension's name with its module name by setting the constraint that they should be the same (and thinking this would make things less confusing for everyone). Though, I can see where you would sometimes want this to be different (e.g. in lab's case). 

I'll move over the jupyter/jupyter_server#222 to discuss further.